modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java

```diff
@@ -208,11 +225,67 @@ public RelDataType convert(SqlDataTypeSpec typeSpec) {
      * @return Validated node, its validated type and type's origins.
      */
     public ValidationResult validateAndGetTypeMetadata(SqlNode sqlNode) {
+        List<SqlNode> selectItems = null;
+        List<SqlNode> selectItemsNoStar = null;
+        SqlNode sqlNode0 = sqlNode instanceof SqlOrderBy ? ((SqlOrderBy)sqlNode).query : sqlNode;
```

Comment:
As far as I know UNION also can be as root node. 
Are you sure we can skip validation for nodes like OrderBy? Perhaps this block should be applied after `SqlNode validatedNode = validator().validate(sqlNode);` on `validatedNode`.

Reply:
Actually, we do not skip validation. This is used just to fix aliases later, after validation